src/reyna/Reyna/Http/HttpClient.cs

```diff
@@ -12,11 +13,13 @@ public sealed class HttpClient : IHttpClient
     {
         public IConnectionManager ConnectionManager { get; set; }
         private IWebRequest webRequest;
+        private IReynaLogger Logger;
 
-        public HttpClient(IConnectionManager connectionManager, IWebRequest webRequest)
+        public HttpClient(IConnectionManager connectionManager, IWebRequest webRequest, IReynaLogger logger)
         {
             this.ConnectionManager = connectionManager;
             this.webRequest = webRequest;
+            Logger = logger;

```

Comment:
Fair point, but Logger is actually ReynaLogger which is created by the UnityContainer so this will never be null. The delegate LogHandler within ReynaLogger is null to begin with so will not log by default. The LogHandler gets set when the caller enables logging ReynaService::EnableLogging so not sure if the exception should be thrown from there but it will need to be caught by the calling application. There are checks in ReynaLogger if (LogHandler != null).